lib/models/room.js

```diff
@@ -119,11 +138,71 @@ function Room(roomId, opts) {
         //   data: <receipt data>
         // }]
     };
+
     this._notificationCounts = {};
+
+    // just a list - *not* ordered.
+    this._timelines = [];
+    this._eventIdToTimeline = {};
+    this._timelineSupport = Boolean(opts.timelineSupport);
+
+    this.resetLiveTimeline();
 }
 utils.inherits(Room, EventEmitter);
 
 /**
+ * Get the live timeline for this room.
+ *
+ * @return {module:models/event-timeline~EventTimeline} live timeline
+ */
+Room.prototype.getLiveTimeline = function() {
+    return this._liveTimeline;
+};
+
+/**
+ * Reset the live timeline, and start a new one.
+ *
+ * <p>This is used when /sync returns a 'limited' timeline.
+ */
+Room.prototype.resetLiveTimeline = function() {
+    var newTimeline;
+
+    if (!this._timelineSupport) {
+        // if timeline support is disabled, forget about the old timelines
+        newTimeline = new EventTimeline(this.roomId);
+        this._timelines = [newTimeline];
+        this._eventIdToTimeline = {};
+    } else {
+        newTimeline = this.addTimeline();
+    }
+
+    if (this._liveTimeline) {
```

Comment:
Clarity that this is expected to be `true` iff `resetLiveTimeout()` is called >1 times would be good (e.g. `//existing timeline from before`). This looks like a bug otherwise.